lib/src/element_type.dart

```diff
@@ -305,8 +305,9 @@ abstract class DefinedElementType extends ElementType {
     if (bound is InterfaceType &&
         !bound.typeArguments.every((t) => t is InterfaceType)) {
       var typeSystem = library.element.typeSystem;
-      return typeSystem.instantiateToBounds2(
-          classElement: bound.element2 as ClassElement,
+      return typeSystem.instantiateInterfaceToBounds(
+          // TODO(srawlins): It sure seems like this will crash on MixinElement.
```

Comment:
Well, then why do you cast it?
`instantiateInterfaceToBounds()` needs `InterfaceElement`, not `ClassElement`.

Reply:
Done!